### PR TITLE
Exam

### PR DESCRIPTION
**Pull request by zainul-realtime**
Hello , I am apologize because i just started this exam in Friday night. I have received test  at Thursday afternoon but I am not completely read at the night I am got tired so I didn't touch . At Friday morning I am fork this repo , I am plan do at this time but many work task created in current job. So I am do at after go home at Friday night.

I am has finished the exam , I believe when we have too much thing in my mind I can do good thing. It's the reason why I just started at Friday night . hopefully that I have done same with the expectation the requirement.

Thank you

### CODE REVIEW

**Comment by diorahman** [comment]
Hi Zainul, thanks for doing this. Could you add unit tests for your submission?

**Comment by zainul-realtime** [comment]
Hi dio 
Okay, I will do it ..
Thank you

**Comment by diorahman** [comment]
Hi @zainul-realtime are you going to add the unit tests or not?

**Comment by zainul-realtime** [comment]
Yes I am going to do it , but I am start do at Monday morning in office hours , in long weekend I am have family event with my family in outside Jakarta ..
I am apologize with in uncomfortable condition

File: test.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+describe('Send SMS', function () {
+    it('should run send sms via helper method', function (done) {
+        setTimeout(function () {
+            helper.sendSms({id:1}, (err, sending)=>{
+              expect(sending.status).to.equal(200)
+              expect(sending.message).to.equal('OK')
+            })
+        }, 600);
+        done();
```
Comment (diorahman) [inline test.js]: This `done()` will be called before that 600 ms timeout

File: package.json [hunk 2]
```diff
@@ -11,10 +11,14 @@
   "dependencies": {
     "async": "^1.5.2",
     "bluebird": "^3.3.4",
+    "chai": "^3.5.0",
```
Comment (diorahman) [inline package.json]: yeah, chai should go to `devDependencies` as well. Did you try to run `npm install` with `--production` flag?

**Comment by zainul-realtime** [comment]
Ohh yap , absolutely chai should be in dev ..
Yeah I am forgot flag --save-dev when I install chai, sorry ..
Yap I take 600ms for get SMS response and waiting when SMS sent ..

**Comment by diorahman** [comment]
> Yap I take 600ms for get SMS response and waiting when SMS sent ..
But that `done()` should be called AFTER the function defined inside the setTimeout is being executed. However in your code it is called BEFORE it.

**Comment by diorahman** [comment]
Anyway thank you for this. We will get back to you after some internal processes.

**Comment by zainul-realtime** [comment]
Ohhhh, i see ..
Yap thank you , for the discussion and give me the chance until this
process.

**Pull request closed without merging**